docs/guide/nameservice/03-types.md

```diff
@@ -1,40 +1,83 @@
 ---
 order: 3
+description: Implement types and methods that operate on the state.
 ---
 
-# Implement Types That Operate On The State
+# Implement Types 
 
 Now that you've defined messages that trigger state transitions, it's time to implement types and methods that operate on the state.
```

Comment:
we can benefit from adding a transition from types > keepers

tell me about this... need to learn more